core/src/main/java/com/novoda/accessibility/ActionsMenu.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Override
+    public ActionMenuItem add(@StringRes int titleRes) {
+        return add(resources.getString(titleRes));
+    }
+
+    @Override
+    public ActionMenuItem add(CharSequence title) {
+        throw new UnsupportedOperationException("Items without IDs explicitly set aren't supported.");
+    }
+
+    @Override
+    public ActionMenuItem add(int groupId, int itemId, int order, @StringRes int titleRes) {
+        return add(groupId, itemId, order, resources.getString(titleRes));
+    }
+
+    @Override
+    public ActionMenuItem add(int groupId, int itemId, int order, CharSequence title) {
+        if (itemId == 0 || menuItemsContainsItemWithId(itemId)) {
+            throw new IllegalArgumentException("itemId for \"" + title + "\" is missing or matches another item in menu.");
+        }
+        ActionMenuItem menuItem = new ActionMenuItem(resources, itemId, title);
+        menuItems.add(menuItem);
+        return menuItem;
+    }
+
+    private boolean menuItemsContainsItemWithId(int itemId) {
+        for (ActionMenuItem menuItem : menuItems) {
+            if (menuItem.getItemId() == itemId) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public SubMenu addSubMenu(CharSequence title) {
+        return addSubMenu(0, 0, 0, title);
+    }
+
+    @Override
+    public SubMenu addSubMenu(@StringRes int titleRes) {
+        return addSubMenu(resources.getString(titleRes));
+    }
+
+    @Override
+    public SubMenu addSubMenu(int groupId, int itemId, int order, @StringRes int titleRes) {
+        return addSubMenu(groupId, itemId, order, resources.getString(titleRes));
+    }
+
+    @Override
+    public SubMenu addSubMenu(int groupId, int itemId, int order, CharSequence title) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int addIntentOptions(int groupId, int itemId, int order, ComponentName caller, Intent[] specifics, Intent intent, int flags, MenuItem[] outSpecificItems) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void removeItem(int id) {
+        ActionMenuItem itemToRemove = null;
+        for (ActionMenuItem menuItem : menuItems) {
+            if (menuItem.getItemId() == id) {
+                itemToRemove = menuItem;
+                break;
+            }
+        }
+        menuItems.remove(itemToRemove);
+    }
+
+    @Override
+    public void removeGroup(int groupId) {
+        List<ActionMenuItem> itemsToRemove = new ArrayList<>();
+        for (ActionMenuItem menuItem : menuItems) {
+            if (menuItem.getGroupId() == groupId) {
+                itemsToRemove.add(menuItem);
+            }
+        }
+        menuItems.removeAll(itemsToRemove);
+    }
+
+    @Override
+    public void clear() {
+        menuItems.clear();
+    }
+
+    @Override
+    public void setGroupCheckable(int group, boolean checkable, boolean exclusive) {
```

Comment:
we could support these at this level, ignoring whether `ActionMenuItem` supports them but will leave for another PR.